packages/crypto/src/blocks/block.ts

```diff
@@ -18,6 +18,15 @@ export class Block implements IBlock {
             throw new BlockSchemaError(data.height, error);
         }
 
+        if (error) {
+            if (isException(value) || data.transactions.some((transaction: ITransactionData) => isException(transaction))) {
+                // Validate again without bailing out on the first error to ensure that all properties get properly converted if necessary
+                validator.validateException("block", data);
+            } else {
+                throw new BlockSchemaError(data.height, error);
```

Comment:
You can remove this, it is never going to be executed. If error is true and the long expression is false then we would have thrown already earlier.
```
if (A && !B) {
  throw E;
}
if (A) {
  if (B) {
    C;
  } else {
    throw E; // this will never be executed
  }
}
```

Or even better: remove the first `if` (lines 14-20) https://github.com/ArkEcosystem/core/pull/3008/files#diff-915b384fb6553933d473ac42c25419c2L14-L20 that code is equivalent to:
```
if (A) {
  if (B) {
    C;
  } else {
    throw E;
  }
}
```

Reply:
Yes, lines 14-20 were intended to be removed.